File: lecture03/02_perfect/template.py
#!/usr/bin/env python3

# Perfect Square

import sys

def is_perfect_square(n):
    # TODO: Use binary search to determine if n is a perfect valid square
    return False

def main():
    for n in map(int, sys.stdin):
        print('Yes' if is_perfect_square(n) else 'No')

if __name__ == '__main__':
    main()

File: lecture03/02_perfect/template.cpp
// perfect_square.cpp: Perfect Square

#include <iostream>

using namespace std;

bool is_perfect_square(long n) {
    // TODO: Use binary search to determine if n is a perfect valid square.
    return false;
}

int main(int argc, char *argv[]) {
    long n;

    while (cin >> n) {
    	cout << (is_perfect_square(n) ? "Yes" : "No") << endl;
    }

    return 0;
}
